### ISSUE TITLE
Key generation fails on GnuPG >=2.1

### ISSUE THREAD

**Issue by satta** [passbolt/passbolt_docker#2]
According to https://www.gnupg.org/documentation/manuals/gnupg/Unattended-GPG-key-generation.html, the following change has been made in newer GnuPG versions:

> For GnuPG versions prior to 2.1, both control statements must be given. For GnuPG 2.1 and later ‘%secring’ is a no-op. 

That means that in setups using these versions `gpg_server_key_private.key` is no longer present and hence the server key generation script fails.

**Comment by stripthis** [comment]
Hi @satta, thank you for reporting this issue, we'll have a look. ref. PASSBOLT-1730

**Comment by satta** [comment]
Thanks! In the meantime will try to manually fetch the private key from the user's `~/.gnupg` and go from there.

**Comment by stripthis** [comment]
Currently "php-gnupg" supports only gnupg version 1 and the version 2 is not currently supported. It should be fixed in the extension version 1.5.
see. https://github.com/php-gnupg/php-gnupg

**Comment by rguillome** [comment]
Hi all,
from the GnuPG sources (doc/DETAILS) 
```    %secring <filename>
	Do not write the key to the default or commandline given
	keyring but to <filename>.  This must be given before the first
	commit to take place, duplicate specification of the same filename
	is ignored, the last filename before a commit is used.
	The filename is used until a new filename is used (at commit points)
	and all keys are written to that file.	If a new filename is given,
	this file is created (and overwrites an existing one).
        GnuPG < 2.1:  Both control statements must be given.
        GnuPG >= 2.1: "%secring" is now a no-op.
```
__%secring is now a no-op__. I'd try with --secret-keyring on the command line, and same result : while public key can be generated by batch, secret key can't with GnuPG >=2.1,
